### DISCUSSION TITLE
No such file or directory error on new software compiling

### DISCUSSION THREAD

**Discussion by asannikov** [Homebrew/discussions#495] [Tap maintenance and Homebrew development] [answered]
hi, i'm trying to add new software using
`brew install --build-from-source jumper`
and getting error:
Error: An exception occurred within a child process:
  Errno::ENOENT: No such file or directory - jumper

the full log:

> ❯ brew install --build-from-source jumper
> Updating Homebrew...
> ==> Auto-updated Homebrew!
> Updated 1 tap (homebrew/core).
> ==> Updated Formulae
> Updated 1 formula.
> 
> ==> Downloading https://github.com/asannikov/jumper/releases/download/v1.5.1/jumper-
> Already downloaded: /Users/User/Library/Caches/Homebrew/downloads/e8d7411eb17de09ec90ef5bc896ed4b12869f371d8a914a987aa9f31c335673f--jumper-darwin-amd64.zip
> Warning: Your Xcode (11.5) is outdated.
> Please update to Xcode 12.3 (or delete it).
> Xcode can be updated from the App Store.
> **Error: An exception occurred within a child process:
>   Errno::ENOENT: No such file or directory - jumper**

my formula:
`/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/jumper.rb`

```
require 'rbconfig'
class Jumper < Formula
  desc "This tool has been created for helping developers out with daily docker routines"
  homepage "https://github.com/asannikov/jumper"
  version "1.5.1"
  license "MIT"
  head "//github.com/asannikov/jumper.git"

  if Hardware::CPU.is_64_bit?
    case RbConfig::CONFIG["host_os"]
    when /mswin|msys|mingw|cygwin|bccwin|wince|emc/
      :windows
    when /darwin|mac os/
      url "https://github.com/asannikov/jumper/releases/download/v1.5.1/jumper-darwin-amd64.zip"
      sha256 "9ed11c20a484b2bf5e549d5ec8cd28c37cbba914ab1a17be8177a38e4b2b6eb1"
    when /linux/
      url "https://github.com/asannikov/jumper/releases/download/v1.5.1/jumper-linux-amd64.tar.gz"
      sha256 "7027c55c0481e6bf7e98c3cabfb7bcef3111fb269cb11eac6439181e5a5296b9"
    when /solaris|bsd/
      :unix
    else
      :unknown
    end
  else
    case RbConfig::CONFIG["host_os"]
    when /mswin|msys|mingw|cygwin|bccwin|wince|emc/
      :windows
    when /linux/
      url "https://github.com/asannikov/jumper/releases/download/v1.5.1/jumper-linux-386.tar.gz"
      sha256 "3685e1822598c22bb2830ffd77d5c9e637f0e7a2a3268905016f1081b6553f98"
    when /solaris|bsd/
      :unix
    else
      :unknown
    end
  end

  system "brew tap tcnksm/ghr"
  depends_on "make" => :run
  depends_on "ghr" => :run
  depends_on 'go' => :build

  GOPATH = ENV["GOPATH"]
  def install
    ENV["GOPATH"] = GOPATH
    system "go get -u github.com/laher/goxc"
    bin.install "jumper"
  end

  test do
      system "jumper"
  end

end
```

what this error means?

**Comment by gromgit** [comment, marked as answer]
It literally means the build process was looking for a file named `jumper`, but couldn't find it.

_Pro tip:_ When installing from source fails, add the `--verbose` and `--debug` flags to help figure out what went wrong.

**Reply by asannikov** [reply]
thank you so much! you are right, the file inside archive was different from jumper. i could not understand the way how brew does it. now i see.
